cpp/src/arrow/python/python-test.cc

```diff
@@ -160,5 +160,41 @@ TEST(BuiltinConversionTest, TestMixedTypeFails) {
   ASSERT_RAISES(UnknownError, ConvertPySequence(list, pool, &arr));
 }
 
+TEST(DecimalMissingValueTest, TestNoneAndNaN) {
```

Comment:
What is the policy for adding tests here rather than in `test_array.py`? It seems writing tests in pure Python is generally easier :-)

Reply:
It's funny you mention that. I agree, but I wanted to be able to step through C++ code in the CLion IDE so I wrote the test here.